plugins/package-managers/maven/src/test/kotlin/utils/DependencyTreeParserTest.kt

```diff
@@ -0,0 +1,376 @@
+            }
+        }
+
+        "skip unknown projects" {
+            val projectNode1 = DependencyTreeMojoNode(
+                "org.ossreviewtoolkit",
+                "module1",
+                "1.2.3-SNAPSHOT",
+                "pom",
+                "",
+                "",
+                listOf(
+                    DependencyTreeMojoNode(
+                        "org.apache.commons",
+                        "commons-configuration2",
+                        "2.11.0",
+                        "jar",
+                        "compile",
+                        ""
+                    )
+                )
+            )
+            val projectNode2 = DependencyTreeMojoNode(
+                "org.ossreviewtoolkit",
+                "module2",
+                "1.2.3-SNAPSHOT",
+                "eclipse-plugin",
+                "",
+                "",
+                listOf(
+                    DependencyTreeMojoNode(
+                        "org.apache.commons",
+                        "commons-lang3",
+                        "3.14.0",
+                        "jar",
+                        "compile",
+                        ""
+                    )
+                )
+            )
+
+            val projectDependencies = parseDependencyTree(
+                inputStreamFor(projectNode1, projectNode2),
+                listOf(createProject("module1"))
+            ).toList()
+
+            projectDependencies shouldHaveSize 1
+            compareNodes(projectNode1, projectDependencies.first())
+        }
+    }
+})
+
+/**
+ * Return an [InputStream] with the JSON representation of the given [projects] in the same way as this would be done
+ * by the dependency tree plugin.
+ */
+private fun inputStreamFor(vararg projects: DependencyTreeMojoNode): InputStream =
+    projects.joinToString("\n") { JSON.encodeToString(it) }.byteInputStream()
+
+/**
+ * Compare a hierarchy of [DependencyTreeMojoNode]s with a hierarchy of [DependencyNode]s given the root nodes
+ * [mojoNode] and [dependencyNode].
+ */
+private fun compareNodes(mojoNode: DependencyTreeMojoNode, dependencyNode: DependencyNode) {
+    dependencyNode.artifact.groupId shouldBe mojoNode.groupId
+    dependencyNode.artifact.artifactId shouldBe mojoNode.artifactId
+    dependencyNode.artifact.version shouldBe mojoNode.version
+    dependencyNode.artifact.extension shouldBe mojoNode.type
+    dependencyNode.artifact.classifier shouldBe mojoNode.classifier
+    dependencyNode.dependency.scope shouldBe mojoNode.scope
+
+    mojoNode.children.size shouldBe dependencyNode.children.orEmpty().size
+    mojoNode.children.zip(dependencyNode.children.orEmpty()).forAll { (mojoChild, child) ->
+        compareNodes(mojoChild, child)
+    }
+}
+
+/**
+ * Create a [MavenProject] with the given [name] and optional [packaging][pckg].
+ */
+private fun createProject(name: String, pckg: String = "pom"): MavenProject =
```

Comment:
Nit: `pckg` sounds strange to me. I'd probably stick to "packaging" and use `this.packaging = packaging`.